src/mongocxx/read_concern.hpp

```diff
@@ -0,0 +1,141 @@
+// Copyright 2015 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <memory>
+
+#include <mongocxx/config/prelude.hpp>
+
+#include <bsoncxx/stdx/string_view.hpp>
+#include <mongocxx/stdx.hpp>
+
+namespace mongocxx {
+MONGOCXX_INLINE_NAMESPACE_BEGIN
+
+class client;
+class collection;
+class database;
+class uri;
+
+///
+/// A class to represent the read concern. Read concern can be set at the client, database, or
+/// collection level. The read concern can also be provided via connection string, and will be
+/// parsed and set on the client constructed for the URI.
+///
+/// For the WiredTiger storage engine, MongoDB 3.2 introduced the readConcern option for replica
+/// sets and replica set shards. The readConcern option allows clients to choose a level of
+/// isolation for their reads. You can specify a readConcern of "majority" to read data that has
+/// been written to a majority of nodes and thus cannot be rolled back. By default, MongoDB uses a
+/// readConcern of "local" which does not guarantee that the read data would not be rolled back.
+///
+/// @TODO link to the docs when they exist.
+///
+class MONGOCXX_API read_concern {
+
+   public:
+    ///
+    /// A class to represent the read concern level.
+    /// @TODO link to the docs when they exist.
+    ///
+    enum class level {
+        k_empty,
```

Comment:
would k_empty still have any meaning if you used optional in the API as suggested above?